src/modules/swap/ModuleSwap.vue

```diff
@@ -1293,9 +1294,8 @@ export default {
             item?.contract &&
             item?.contract?.toLowerCase() === value?.contract?.toLowerCase()
           )
-            return item;
+            value = foundToken[0];
```

Comment:
You need to `return item` here then outside of the filter call set `value = foundToken[0];`